tests/experimental/nn/test_multiprocess_pipe.py

```diff
@@ -257,3 +256,54 @@ def multi_input_multi_output_layers(devices):
             opt.step(context_id)
     losses = [l.to_here() for l in losses]
     assert losses[0] > losses[1], f"{losses[0]} !> {losses[1]}"
+
+
+# A test for extracting the same graph as in test multi_input_multi_output_layers automatically
+class ShardedLinearLayer(nn.Module):
+    def __init__(self, input_device, shard_devices, output_device):
+        super().__init__()
+        self.split = RemoteModule(input_device, SplitTensors, (), {})
+        self.linear_layers_2 = nn.ModuleList(
+            [
+                RemoteModule(shard_devices[0], nn.Linear, (2, 2), {}),
+                RemoteModule(shard_devices[1], nn.Linear, (2, 2), {}),
+            ]
+        )
+        self.concatenate = RemoteModule(output_device, ConcatenateTensors, ())
+
+    def forward(self, input):
+        shards = self.split(input)
+        shards = [self.linear_layers_2[i](shards[i]) for i in range(2)]
+        return self.concatenate(*shards)
+
+
+@rpc_test(world_size=2)
+@pytest.mark.parametrize("devices", DEVICES)
+def auto_graph_extract(devices):
+    from fairscale.experimental.nn.distributed_pipeline.trace import make_graph
```

Comment:
Move import to the top of the file? unless there is an issue?

Reply:
trace can not be imported for older versions of pytorch (where this test is disabled).